### PR TITLE
#2629: update startup logic docs

### PR DESCRIPTION
**Pull request by cuserox**
### Why does this PR exist?

Closes [#2729](https://github.com/tokens-studio/figma-plugin/issues/2729) 

Previous description of the plugin startup logic for the plugin was outdated. This improves the information as it stands now and some formatting improvements.

### What does this pull request do?

As part of several jam sessions ([board](https://www.figma.com/file/4vAZ27ycAiNNh3ohd5yNM9/Tokens-Studio-Plugin%3A-startup-logic-%2B-misc?type=whiteboard&node-id=0%3A1&t=CvMW2I7CYdmsZ8Ry-1)), the startup logic was revisited to get a better overview on how it works now, and aid in planning improvements.

### Testing this change

Preview the updated [startup-logic.md](https://github.com/tokens-studio/figma-plugin/blob/2729-startup-logic-docs/developer-knowledgebase/startup-logic.md) file ✨

### CODE REVIEW

**Review by LukeFinch — APPROVED** [review]
This is awesome!

File: developer-knowledgebase/startup-logic.md [hunk 1]
```diff
@@ -1,52 +1,41 @@
...
+
+![Startup logic for the backend](./images/startup-backend.jpg)
+
+### Frontend
+
+Once the precursing information has been acquired, `index.tsx` is in charge of displaying the actual plugin view. Its responsibilities include:
+* Initializing our tracking tools: Mixpanel and Sentry
+* Instantiating the asynchronous messaging channel for the part of code communicating with our 'backend' (`ReactInstance`)
+* Registering the following events and handlers:
+   * `GET_THEME_INFO`: to know the last active theme of the user
```
Comment (macintoshhelper) [inline developer-knowledgebase/startup-logic.md:29]: Maybe worth mentioning that this message is called from the UI/front-end instance, then a message listener is created with a promise callback that gets resolved with the data, and destroys the listener once the controller/back-end receives the message and sends a message with the same type `GET_THEME_INFO` together with the requested data.

File: developer-knowledgebase/startup-logic.md [hunk 2]
```diff
@@ -1,52 +1,41 @@
...
+   * `STARTUP`: finds the container in the document to render the root of the React application. Wrapped with providers to allow Sentry error tracking, Redux state handling and tooltips
```
Comment (macintoshhelper) [inline developer-knowledgebase/startup-logic.md:30]: Could be worth mentioning that `STARTUP` is called from the controller in `init.ts` and is then resolved by `AsyncMessageChannel`, which then calls the registered handler in `startup.tsx` via a callback, to trigger an initial render.

File: developer-knowledgebase/startup-logic.md (hunk 1 above)
Comment (cuserox) [reply developer-knowledgebase/startup-logic.md:29]: I show this flow of info in the simplified frontend graph ([here](https://www.figma.com/file/4vAZ27ycAiNNh3ohd5yNM9/%F0%9F%97%BA%EF%B8%8F-Startup-logic-brainstorm?type=whiteboard&node-id=122%3A886&t=rTHPN7YWulAwqYK1-1)) 🗺️ 

Ideally, I envisage the extra information you're referring to to live in its own `async-message-channel.md` doc, to be filled up as we're planning on [refactoring it](https://github.com/tokens-studio/figma-plugin/issues/2744#issue-2289927040) (and replacing it with something more stable!)

**Review by macintoshhelper — APPROVED** [review]
LGTM 🙂

**Pull request merged**